### PR TITLE
Adds the Monk mrtial art as a null-rod variant. Code by goofball,

### PR DESCRIPTION
**Pull request by GuyonBroadway**
stolen by guyon.

:cl: Iamgoofball and GuyonBroadway
add: The Chaplain has a new Martial Art available in replacement of his Null Rod. Look for the Monk Manual.
/:cl:

[why]: Something for non antag crew to do. Monks can eventually train disciples after (much) training. 

But a wet floor tile will cuck you all the same. 

Synergises VERY nicley with the gloves of the north star (video pending)

Modularisation may be VERY messy, any suggestions

### CODE REVIEW

**Comment by Pyko1** [comment]
+9001

**Comment by ghost** [comment]
let the age of mcDojo begin

**Comment by ghost** [comment]
everyone's kung fu fighting

**Comment by praisenarsie** [comment]
lol you are so stupid

**Comment by GuyonBroadway** [comment]
One of the reasons tg hate the PR was because it forced you to become the human species.On that logic I say we're more accepting.

**Comment by ghost** [comment]
WARNING WARNING NON HEADADMIN OPINION!!!!

I dislike the nullrod being non cosmetic and I feel powers shouldn't be opt in like this. Nullrods should supplement Chaplain RP gimmicks rather than be selected based on a power playing level.

**Comment by Manly-Dwarf** [comment]
@McBawbaggings The Chaplain has nothing to do when the gamemode isn't Cult. Doing prayer gimmicks relies entirely on an admin being online, and said admin not being a trialmin, as well as that admin not having a pole up their ass/a shit-ton of ahelps to deal with. While I know you guys lie about how many ahelps you actually have to deal with, sometimes admins are just not able to do gimmicks in the round.

Why should the Chaplain be entirely bound to roleplay? It'd probably be for the better if the Chaplain could have _something_ to really call his own that isn't bound to admins.

**Comment by ghost** [comment]
really cool

**Comment by GuyonBroadway** [comment]
Imma slap a WIP tag on this for now as its goofcode and a few months old, I just did my best to get it to compile, now we need to make it no shitty.

**Comment by praisenarsie** [comment]
again this is too op and too stupid, it's like gangs but crack shoved up your bumhole while eating the Carolina reaper

**Comment by ExpAntique** [comment]
+1, I want to start a monk dojo dammit

 edit: /tg/ crybabies ended closing goof's for bs reasons, let us not do the same

**Comment by Manly-Dwarf** [comment]
@praisenarsie 
>melee only
>requires a ton of leveling up which was deliberately designed to take forever
>any good attack like stunning punches is RNG and has limited uses
>"OP"

**Comment by praisenarsie** [comment]
nigga u can literally attack someone for days while they heal at the same time and get easy levels

**Comment by Manly-Dwarf** [comment]
That doesn't change shit. It still takes forever, and the attacks are still RNG. You'd be better off just using a stunbaton.

**Comment by GuyonBroadway** [comment]
Or you could pick gloves of the north star and sleeping carp and murder people almost as fast from the get go and be ranged immune and have a garunteed disarm to boot.

File: code/__DEFINES/~hippie_defines/maths.dm [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Real modulus that handles decimals
+#define ROLL_DICE(amt, faces) (rand(0, (faces-1)*amt) + amt)
```
Comment (GuyonBroadway) [inline code/__DEFINES/~hippie_defines/maths.dm]: You know whats funny, this was in the goof PR but its not used. I should really remove this as its unused.

**Comment by GuyonBroadway** [comment]
Ok this is ready for some other coder to take a gander and revel in the horror that is goofcode.

**Comment by Carbonhell** [comment]
perhaps give it as default to chaplains instead of it being tied to which nullrod you choose?

**Comment by ghost** [comment]
how about we got dominions 5 and make it work like the cook's CQC,  only in the chapel?  buuuut the chaplain can sanctify other rooms to be able to fight in them if he prays enough/something else to spread his God's dominion?

**Comment by GuyonBroadway** [comment]
Eh... its kinda strong for a non antag thing to be getting just for free, the tradeoff is he no longer HAS his nullrod which is very effective vs cults. 

Plus when combo'd with gloves of the north star a high level monk IS a death machine, he could couple that with the staff of protection which would make him near invincible when coupled with the monks dodge/block chance.

**Comment by ghost** [comment]
Head admin from TG, I really promise you this goes extremely badly TRUST me

**Comment by praisenarsie** [comment]
if only i could close this.

well i could but carbonhell will whine and boot me off

**Comment by GuyonBroadway** [comment]
This is hippie, we run on bad ideas and autistic screeching.

**Comment by praisenarsie** [comment]
shut up powergamer!!

**Comment by Iamgoofball** [comment]
@Armhulen actually itd probably work out very well here

**Comment by Iamgoofball** [comment]
@GuyonBroadway this doesnt stack with other martial arts you fucking nerd

**Comment by Pyko1** [comment]
we can just always nuke it to oblivion if its get too bad.

**Comment by GuyonBroadway** [comment]
@Iamgoofball Gloves of the north star isn't a martial art. :^)

**Comment by FantasticFwoosh** [comment]
This was powergamed on tg to crit people with lucky flurries by rigorously training punching, so test merge and give it a whirl first.

But there can only be one heir to the sacred & secret art of Hokutu No Ken.

**Comment by GuyonBroadway** [comment]
People on hippie are too ADHD to put the time in to grind levels and are too allergic to co-operation to allow a monk to beat the shit out of you for any length of time in the hopes that they "might" get to be a disciple.

**Comment by Pyko1** [comment]
I mean this will hyst be an excuse to murder chaplain since "he was an obvious antag"

**Comment by Carbonhell** [comment]
cautious +1, if it goes bad we'll just remove it

**Comment by earthmoonjune** [comment]
+1

**Comment by Jujumatic** [comment]
this still works fine after the upstream?

**Comment by GuyonBroadway** [comment]
It needed a parity change but it works now.

**Comment by Jujumatic** [comment]
le oof this just died

**Comment by GuyonBroadway** [comment]
Le Un-Oof dis works again.

**Comment by GuyonBroadway** [comment]
So yeah this has been recently re-tested and is good to go.

**Review by Jujumatic — APPROVED** [review]
looks good though i'm concerned it might end up being spammed by powergamers. maybe some balance concerns further down the line? but definitely looks good for testing on live at the very least

**Pull request merged**